src/alogamous/line_count_analyzer.py

```diff
@@ -0,0 +1,12 @@
+from alogamous import analyzer
+
+
+class LineCountAnalyzer(analyzer.Analyzer):
+    def __init__(self):
+        self.line_counter = 0
+
+    def read_log_line(self, _):
+        self.line_counter += 1
+
+    def report(self, out_stream):
+        out_stream.write("Number of log lines: " + str(self.line_counter))
```

Comment:
I think you need to add a newline `"\n"` character at the end of this line.

Either that, or the `analyze_log_stream` function needs to add a newline after it calls `report` for an Analyzer. Which do you prefer here @laurenassour?

Reply:
Lets add it to `analyze_log_stream` so we don't have to remember it every time we make a new analyzer!